vsphere/vsphere.go

```diff
@@ -767,10 +764,16 @@ func (ops *vsphereOps) getDatastoreToUseInStoragePod(
 	}
 
 	spec.DeviceChange = deviceChange
-	// It's best effort locking. If we didn't get a lock no harm done.
-	storeLock, lockErr := ops.dsLock.LockWithKey(ops.cfg.VMUUID, storagePod.Name())
+	var (
+		storeLock *store.Lock
+		lockErr error
+	)
+	if ops.dsLock != nil {
```

Comment:
Should we explicitly error out if dsLock is nil? 
I know that there is no code that invokes this function without initializing store. I am trying to future proof it to ensure `recommendDatastore` is always called with store initialized.